aspnet/migration/aspnetcoremigration.rst

```diff
@@ -0,0 +1,190 @@
+
+The tool needs now to distinguish portable apps from standalone apps to be able to write the ``web.config`` file correctly depending on the application type that is being published. This required adding a new, mandatory parameter ``--framework`` that tells the tool what framework the application was published for.
+
+since HttpPlatformModule was replaced with AspNetCoreModule
+the web.config created by the publish-iis tool now configures IIS to use AspNetCoreModule instead of HttpPlatformHandler to forward requests to Kestrel
+
+The code snippet below shows how to configure the new publish-iis tool in project.json file
+
+.. code-block:: JavaScript
+
+"tools": {
+  "Microsoft.AspNetCore.Server.IISIntegration.Tools": {
+    "version": "1.0.0-*",
+    "imports": "portable-net45+wp80+win8+wpa81+dnxcore50"
+  }
+},
+"scripts": {
+  "postpublish": "dotnet publish-iis --publish-folder %publish:OutputPath% --framework %publish:FullTargetFramework%"
+}
+
+Additionally, you must turn on server garbage collection in ``project.json`` or ``app.config`` when on full .NET framework.
+
+Changes in MVC
+--------------
+
+- Set option in ``project.json`` to preserve the compilation context if you are doing views.
+- You no longer need to reference the Tag Helper package, it's now referenced by ``Razor`` by default.
+
+There are changes that simplify controller discovery:
+There is a new ``Controller`` attribute that can be used to mark a class and their descendants as controllers.
+Classes whose name doesn't end in ``Controller`` and derive from a base class that ends in ``Controller`` are no longer considered controllers. In this scenario the ``[Controller]`` attribute must be applied to the ``Controller`` class itself or to the base class.
+
+We now consider a type to be a controller if all of the following rules apply:
+- The type is a public, concrete, non open generic class.
+- [NonController] is not applied to any type of the hierarchy.
+- The type name ends with ``Controller``, or if the ``[Controller]`` attribute is applied to the type or to one of its ancestors.
+- It's important to note that if [NonController] is applied anywhere in the type hierarchy the discovery conventions will never consider that type or its descendants to be a controller. ``[NonController]`` takes precedence over ``[Controller]``.
+
+
+ViewComponents changes
+----------------------
+
+-The Sync APIs have been removed.
+-``InvokeAsync`` takes an anonynmous object instead of params.
+-``web`` is no longer a thing, use dotnetrun or ``dotnet <dll name>`` at the command prompt instead.
+
+Mapping commands to tools
+-------------------------
+
+The following commands are now tools: Secret manager, watch, sqlcache. You can configure these in the ``tools`` section in the ``project.json`` file found in the root of your project. 
```

Comment:
The following DNX commands have been replaced with the following .NET Core tools:
`<RC1 vs RC2 table here>`